### PR TITLE
refactor: improve Defaults.Key

### PR DESCRIPTION
**Pull request by tisfeng**
Close https://github.com/tisfeng/Easydict/issues/497

This PR does not include functional changes, it reduces a large amount of duplicate code related to LLMStreamService configuration, the main content is as follows:

- Remove all static `Defaults.Keys` of LLMStreamService, replacing them with dynamic keys related to serviceType.
- Use the instance method `configurationListItems()` to generate the configuration view.

This PR has many changes, hoping it doesn't have any breaking changes, please review carefully.

### CODE REVIEW

**Review by Jerry23011 — DISMISSED** [review]

**Review by AkaShark — APPROVED** [review]

File: Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift [hunk 1]
```diff
@@ -8,64 +8,44 @@
...
 }
+
+// MARK: - StoredKey
+
+enum StoredKey: String {
+    case serviceUsageStatus
+    case translation
+    case dictionary
+    case sentence
+    case supportedModels = "AvailableModels" // save in String: "gpt-3.5, gpt-4"
```
Comment (phlpsong) [inline Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:39]: The default string will be lowered case, maybe make them consistent here?

File: Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift [hunk 2]
```diff
@@ -8,64 +8,44 @@
...
-    storedKey(EZAPIKey, serviceType: serviceType)
-}
-
-func endpointStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZEndpointKey, serviceType: serviceType)
-}
-
-func nameStoredKey(_ serviceType: ServiceType) -> String {
-    storedKey(EZNameKey, serviceType: serviceType)
+    "EZ" + serviceType.rawValue + key.rawValue.capitalizeFirstLetter() + "Key"
```
[the remark this answers is not in the document]
Comment (tisfeng) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:15]: I don't understand, what exactly do you mean?

File: Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift (hunk 1 above)
Comment (tisfeng) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:39]: The key here needs to be compatible with the previous constants `EZOpenAIAvailableModelsKey` and `EZCustomOpenAIAvailableModelsKey`. 

Do you have any other ideas, please give details.

File: Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (phlpsong) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:15]: If we want to have multi instances of service, it needs to be distinguished on the configuration key name.

For example, image we have two services OpenAIService A and B, we shall have different configurations for each of them. 

For the key name, we shall have maybe `EZ_OpenAI_E_Key_A` and `EZ_OpenAI_E_Key_B` (use `name` key here) for each of them.  But the current implementation cannot achieve this.

[the remark this answers is not in the document]
Comment (tisfeng) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:15]: I see, later we can add a parameter like id, what do you think?

```swift
func storedKey(_ key: StoredKey, serviceType: ServiceType, id: String? = nil) -> String {
    // This key should be compatible with existing OpenAI config keys
    // EZOpenAIServiceUsageStatusKey
    // EZOpenAIDictionaryKey
    // EZOpenAIDictionary_ID_Key

    var identifier = ""
    if let id {
        identifier = "_" + id + "_"
    }
    return "EZ" + serviceType.rawValue + key.rawValue.capitalizeFirstLetter() + identifier + "Key"
}
```

[the remark this answers is not in the document]
Comment (phlpsong) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:15]: Yes, it needs to be changed according to multi instances support later.

File: Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift (hunk 1 above)
Comment (phlpsong) [reply Easydict/Swift/Feature/Configuration/DefaultsStoredKey.swift:39]: OK, got it, this should be fine.

**Comment by phlpsong** [comment]
It looks like the `Use Model` config is not working as expected now:

Screenshot 2024-07-06 at 18 25 43

**Comment by tisfeng** [comment]
> It looks like the `Use Model` config is not working as expected now:
> 
> Screenshot 2024-07-06 at 18 25 43

How do you trigger this issue? Can you reproduce it?

**Comment by phlpsong** [comment]
Yes, it seems pretty easy to reproduce. Some behavior occurs on my another device too.

**Comment by tisfeng** [comment]
ok, I fixed it [b2485c3](https://github.com/tisfeng/Easydict/pull/599/commits/b2485c32ec10269cf84685b51a1eb4eb87d33814) , please try it again.

**Comment by phlpsong** [comment]
> ok, I fixed it [b2485c3](https://github.com/tisfeng/Easydict/pull/599/commits/b2485c32ec10269cf84685b51a1eb4eb87d33814) , please try it again.

The issue in  `BuiltInService` is fixed, but OpenAI and CustomAI service has the similar issue, `Use Model` not changed after input supported models.

Screenshot 2024-07-06 at 22 50 12

**Comment by tisfeng** [comment]
ok, I will take a look later 🥲

**Comment by tisfeng** [comment]
I tried to reset OpenAI `modelKey`, `validModels` and `supportedModels` default value, then print `validModels` seems good.

I tested on the service configuration page, and `Use Model` follows the change of `Supported Models`, everything works fine.

I've checked the code and didn't find any issues, please test carefully or provide more problem information.

image

**Comment by phlpsong** [comment]
I tried to reset these keys, but after input new values in supported models, the same issue still exists.


[screen-capture.webm](https://github.com/tisfeng/Easydict/assets/103433299/5cf26b95-6ab4-4135-839e-225d8d11873d)

**Comment by tisfeng** [comment]
After checking, I found that if I turn off OpenAI service, modifying the `Supported Models` at this time will not cause the `Use Model` to change, which is because we currently only monitor the services turned on in the query window.

This is a bug, and I will fix it later.

@phlpsong Is this the situation you encountered?

https://github.com/tisfeng/Easydict/assets/25194972/56778b63-f29b-439e-a51f-a2e446ac209c

**Comment by phlpsong** [comment]
> After checking, I found that if I turn off OpenAI service, modifying the `Supported Models` at this time will not cause the `Use Model` to change, which is because we currently only monitor the services turned on in the query window.
> 
> This is a bug, and I will fix it later.
> 
> @phlpsong Is this the situation you encountered?
> 
>  iShot_2024-07-07_21.25.36.mp4

Not exactly the same, I tried to enable and disable the service, the Use Model still not get updated.


https://github.com/tisfeng/Easydict/assets/103433299/1655a0ca-6a8e-4986-b57b-0007010c84cc


I also tried to reset these models first and give a try:


https://github.com/tisfeng/Easydict/assets/103433299/9693b6ee-33dc-4579-92dc-f9fe404c7386

**Comment by tisfeng** [comment]
No, what I mean is, you must first display a query window, and then turn on OpenAI service, only then will it start listening for changes in Defaults.

**Comment by phlpsong** [comment]
> it

OK, it's working. 

Screenshot 2024-07-08 at 08 44 56

**Comment by tisfeng** [comment]
This bug is not easy to resolve, let's merge this PR first, and I'll fix it in another PR later.

**Comment by phlpsong** [comment]
> This bug is not easy to resolve, let's merge this PR first, and I'll fix it in another PR later.

ok, np

**Review by phlpsong — APPROVED** [review]

**Pull request merged**